### PR TITLE
Init ACRA when starting commons application

### LINKED ISSUE #1882: ACRA crash reports do not work again in 2.8.2
**Summary:** 

When the app crashes, there should be an option to "send feedback" so that users can send their stack trace to us along with comments about what they were doing at the time of the crash.

We already have ACRA for this purpose which used to work, but stopped working again at some point.

If anyone takes this up, please ensure that the logs are sent to the PRIVATE google group, not the public one, due to privacy concerns.

**Steps to reproduce:** 

Trigger a crash on 2.8.2. Not sure how far back this problem goes.

**Add System logs:**

**Device and Android version:** 

8.0 Samsung Galaxy s7
 
 **Commons app version:** 

2.8.2 Play Store version 

**Would you like to work on the issue?**

Pref not

### PR DESCRIPTION
**Pull request by albendz**
**Description (required)**

Fixes #1882 

What changes did you make and why?
Add ACRA initialization, as was previously present in working branches.

**Tests performed (required)**

Tested Debug on emulator Nexus 5X with API level 25. Forced uncaught runtime exception to trigger crash report dialog.

**Report Produced**
`USER_COMMENT=albendz testing crashes
ANDROID_VERSION=7.1.1
APP_VERSION_NAME=2.9.0-debug-acra_fix~bbe5d089
BRAND=google
PHONE_MODEL=Android SDK built for x86
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.beta/fr.free.nrw.commons.AboutActivity}: java.lang.RuntimeException
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6119)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: java.lang.RuntimeException
	at fr.free.nrw.commons.AboutActivity.onCreate(AboutActivity.java:76)
	at android.app.Activity.performCreate(Activity.java:6679)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
	... 9 more
java.lang.RuntimeException
	at fr.free.nrw.commons.AboutActivity.onCreate(AboutActivity.java:76)
	at android.app.Activity.performCreate(Activity.java:6679)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6119)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)

`

Need help? See https://support.google.com/android/answer/9075928

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java [hunk 1]
```diff
@@ -91,6 +91,7 @@
     @Override
     public void onCreate() {
         super.onCreate();
+        ACRA.init(this);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Acra is also inited on line number 122.

Comment (albendz) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Do you think moving it fixes the missing dialogs? If so, I'll remove the one on 122 which I missed.

Comment (albendz) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Played around a bit: using the initAcra method does not work.

Comment (albendz) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Yeah, having this Acra.init() line results in the dialog asking for feedback again after a crash. Using the initAcra() after the  onCreate does not fix it.

Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Great! In that case lets remove initAcra function. Also, can you try sending logs to check if the reports are getting generated correctly

**Comment by misaochan** [comment]
Have not had time to test this yet, but looks good to me. Happy to merge pending successful tests. :)

**Comment by domdomegg** [comment]
Tested `2.9.0-debug-pr-2091~da0af75c` on a `Samsung Galaxy S6 7.1.2`, with API level `25`.

Everything appears to work. Forcing a crash gives this output:

```
USER_COMMENT=Example description of what I was doing.
ANDROID_VERSION=7.1.2
APP_VERSION_NAME=2.9.0-debug-pr-2091~da0af75c
BRAND=Samsung
PHONE_MODEL=SM-G920F
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Test RuntimeException to demo ARCA
at fr.free.nrw.commons.contributions.MainActivity.onBackPressed(MainActivity.java:243)
at android.app.Activity.onKeyUp(Activity.java:2726)
at android.view.KeyEvent.dispatch(KeyEvent.java:2717)
at android.app.Activity.dispatchKeyEvent(Activity.java:3030)
at android.support.v7.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:534)
at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
at android.support.v7.app.AppCompatDelegateImplBase$AppCompatWindowCallbackBase.dispatchKeyEvent(AppCompatDelegateImplBase.java:316)
at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:322)
at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4337)
at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4308)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3859)
at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3912)
at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3878)
at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4005)
at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3886)
at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4062)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3859)
at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3912)
at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3878)
at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3886)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3859)
at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3912)
at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3878)
at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4038)
at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:4199)
at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:2400)
at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:1964)
at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:1955)
at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:2377)
at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:141)
at android.os.MessageQueue.nativePollOnce(Native Method)
at android.os.MessageQueue.next(MessageQueue.java:323)
at android.os.Looper.loop(Looper.java:136)
at android.app.ActivityThread.main(ActivityThread.java:6186)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

**Pull request merged**